pkg/snapstore/abs_snapstore.go

```diff
@@ -72,35 +74,41 @@ func NewABSSnapStore(config *brtypes.SnapstoreConfig) (*ABSSnapStore, error) {
 		Retry: azblob.RetryOptions{
 			TryTimeout: downloadTimeout,
 		}})
-	u, err := url.Parse(fmt.Sprintf("https://%s.%s", storageAccount, brtypes.AzureBlobStorageHostName))
+
+	absURI := brtypes.AzureBlobStorageGlobalURI
+	// configuration provides an override for Azure's URI
+	if overrideURI != nil {
+		absURI = *overrideURI
+	}
+	u, err := url.Parse(fmt.Sprintf("https://%s.%s", storageAccount, absURI))
```

Comment:
Using `URI` is semantically incorrect as URI identifies a resource and URL locates a resource. So here you call a domain a URI and then use that to create a URL. Why i call it a domain is because originally we wanted the consumers to specify the domain and not the URL. Therefore what you do is a bit weird. Are you expecting the end user to provide a fully formed URL in the secret if not then are you validating that? If not then who is validating it?

Reply:
As per our discussions, we have decided that the consumer must provide the fully qualified URL for the storage account, where the URL will be of the form `http[s]://<storage-account>.<domain>/`.

Thanks for your suggestion.